### PR TITLE
Use more precomputation for price to tick

### PR DESCRIPTION
**Pull request by ValarDragon**
This PR uses pre-computation for price to tick, and (at least to me) feels like it simplifies the code.

This also surfaces questions for the bucket index definition that I think we should be handling in a follow-up PR

### CODE REVIEW

File: x/concentrated-liquidity/types/constants.go [hunk 1]
```diff
@@ -8,20 +8,22 @@ import (
 
 const (
 	// Precomputed values for min and max tick
-	MinTick, MaxTick int64 = -162000000, 342000000
+	MinTick, MaxTick              int64 = -162000000, 342000000
+	ExponentAtPriceOne            int64 = -6
+	ConcentratedGasFeeForSwap           = 10_000
+	BaseGasFeeForNewIncentive           = 10_000
+	BaseGasFeeForInitializingTick       = 10_000
```
Comment (ValarDragon) [inline x/concentrated-liquidity/types/constants.go:15]: Driveby, move more things to constants

File: x/concentrated-liquidity/math/tick.go [hunk 2]
```diff
@@ -168,51 +168,45 @@ func powTenBigDec(exponent int64) osmomath.BigDec {
...
+	// The number of ticks that need to be filled by our current spacing is
+	// (price - geoSpacing.initialPrice) / geoSpacing.additiveIncrementPerTick
+	priceInThisExponent := osmomath.BigDecFromSDKDec(price.Sub(geoSpacing.initialPrice))
+	ticksFilledByCurrentSpacing := priceInThisExponent.Quo(geoSpacing.additiveIncrementPerTick)
+	// we get the bucket index by:
+	// * taking the bucket index of the smallest price in this tick
+	// * adding to it the number of ticks "completely" filled by the current spacing
+	// the latter is the truncation of the division above
+	// TODO: This should be rounding down?
+	tickIndex = geoSpacing.initialTick + ticksFilledByCurrentSpacing.SDKDec().RoundInt64()
```
Comment (ValarDragon) [inline x/concentrated-liquidity/math/tick.go:210]: If we do roundInt64, we are defining a bucket `b` to be the range `TickToPrice(b - .5)`, `TickToPrice(b + .5)`

If we want the definition (which I think we intend), we should be doing bucket `b` is the range `TickToPrice(b)` to `TickToPrice(b + 1)`. That would be rounding down. If we agree, lets change that in a subsequent PR

File: x/concentrated-liquidity/math/precompute.go [hunk 3]
```diff
@@ -17,8 +18,67 @@ var (
...
+
+func buildTickExpCache() {
+	// build positive indices first
+	maxPrice := sdkOneDec
+	curExpIndex := int64(0)
+	for maxPrice.LT(types.MaxSpotPrice) {
+		tickExpCache[curExpIndex] = &tickExpIndexData{
+			// price range 10^curExpIndex to 10^(curExpIndex + 1). (10, 100)
+			initialPrice:             sdkOneDec.Mul(sdkTenDec.Power(uint64(curExpIndex))),
+			maxPrice:                 sdkOneDec.Mul(sdkTenDec.Power(uint64(curExpIndex + 1))),
```
Comment (p0mvn) [inline x/concentrated-liquidity/math/precompute.go:59]: Why do we multiply 1 here?

Comment (p0mvn) [reply x/concentrated-liquidity/math/precompute.go:59]: Also, any reason for not using `PowTenInternal` or `powTenBigDec` here?

File: x/concentrated-liquidity/math/precompute.go [hunk 4]
```diff
@@ -17,8 +18,67 @@ var (
...
+
+var tickExpCache map[int64]*tickExpIndexData = make(map[int64]*tickExpIndexData)
+
+func buildTickExpCache() {
+	// build positive indices first
+	maxPrice := sdkOneDec
+	curExpIndex := int64(0)
+	for maxPrice.LT(types.MaxSpotPrice) {
+		tickExpCache[curExpIndex] = &tickExpIndexData{
+			// price range 10^curExpIndex to 10^(curExpIndex + 1). (10, 100)
```
Comment (p0mvn) [inline x/concentrated-liquidity/math/precompute.go:57]: is (10, 100) a range notation implying exclusiveness?

File: x/concentrated-liquidity/math/tick.go (hunk 2 above)
Comment (p0mvn) [reply x/concentrated-liquidity/math/tick.go:210]: The definition makes sense to me. However, I don't understand why many inverse relationship tests break if we make this change.

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/concentrated-liquidity/math/precompute.go (hunk 4 above)
Comment (ValarDragon) [reply x/concentrated-liquidity/math/precompute.go:57]: Ah, I don't have a good sense of the inclusiveness / exclusiveness that we actually use, here its not

File: x/concentrated-liquidity/math/precompute.go (hunk 3 above)
Comment (ValarDragon) [reply x/concentrated-liquidity/math/precompute.go:59]: oh true lol
